src/train_cmd.cpp

```diff
@@ -1664,15 +1664,14 @@ void ReverseTrainSwapVeh(Train *v, int l, int r)
 	}
 }
 
-
 /**
  * Check if the vehicle is a train
  * @param v vehicle on tile
- * @return v if it is a train, nullptr otherwise
+ * @return true if v is a train
  */
-static Vehicle *TrainOnTileEnum(Vehicle *v, void *)
+static bool TrainOnTileEnum(const Vehicle *v)
```

Comment:
Maybe rename to `IsTrain`? The whole 'OnTile' bit seems to be a relic from the signal code this seems to be copied from.